### ISSUE TITLE
[Table]: WCAG and other issues

### ISSUE THREAD

**Issue by ivarkallejarv** [TEHIK-EE/tedi-design-system#804]
## Problem
- Add sorting and/or filtering to table. JAWS reads out "Filtreeri, button" and "Sorteeri kasvavalt", which gives no indication as to what you can filter by, i.e. filter by name or etc etc (column name most likely, not 100% sure what the standard is here)
![Image](https://github.com/user-attachments/assets/e28f77cf-115a-4b75-97a7-5dbc12d1b0ed)

- Add items to table, and also filter if needed. Upon state changes, pagination doesn't read out number of results or other necessary information
![Image](https://github.com/user-attachments/assets/c18e79a7-62a2-4df2-ac0a-dd98c48c2902)
